src/azul/service/catalog_controller.py

```diff
@@ -41,14 +40,13 @@ def list_catalogs(self) -> schema.object(
                 catalog.name: {
                     'internal': catalog.is_internal,
                     'atlas': catalog.atlas,
-                    'plugins': [
-                        {
-                            'type': plugin_type,
-                            'name': plugin,
+                    'plugins': {
+                        plugin_type: {
+                            **plugin,
                             **self._plugin_config(plugin_type, catalog.name)
                         }
-                        for plugin_type, plugin in catalog.plugins.items()
-                    ]
+                        for plugin_type, plugin in sorted(catalog.plugins.items())
```

Comment:
> The order of the dictionary can change if someone changes the order of the defined plugins

For the record, that's deterministic and wouldn't require sorting. Like I said in other PRs, we almost always want to eliminate the non-determinisms as opposed to working around them with sorting. Also, we need to be sure we understand the difference between non-determinisms and ordering sensitivity.

> but because of how dictionaries are accessed I think this is a non-issue.

It's the same issue. Changing the order of entries in a dict is equivalent to changing the order of items in a list. So the dict-of-dicts version is still sensitive to ordering changes in the configuration. But that's neither a non-determinism nor is it a problem in this case.